src/workerd/api/streams/internal.c++

```diff
@@ -202,7 +198,14 @@ class AllReader final {
       }
     }
 
-    if (option == ReadOption::NULL_TERMINATE) {
+    if (option != ReadAllTextOption::NONE) {
+      // Strip UTF-8 BOM if requested
+      if (option == ReadAllTextOption::NULL_TERMINATE_AND_STRIP_BOM && parts.size() > 0 &&
+          hasUtf8Bom(parts[0].asBytes())) {
+        parts[0] = kj::heapArray<T>(parts[0].slice(UTF8_BOM_SIZE));
```

Comment:
Ideally we'd be able to skip the additional allocation here, especially since we're copying again immediately below. Can this be modified such that instead of modifying `parts[0]` here, the `copyInto` can be told to ignore the first three bytes of `parts[0]`?